api/v1beta1/conversion.go

```diff
@@ -16,14 +16,187 @@ limitations under the License.
 
 package v1beta1
 
-// Hub marks CloudStackCluster as a conversion hub.
-func (*CloudStackCluster) Hub() {}
+import (
+	"context"
+	"fmt"
+	"github.com/apache/cloudstack-go/v2/cloudstack"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	conv "k8s.io/apimachinery/pkg/conversion"
+	"sigs.k8s.io/cluster-api-provider-cloudstack/api/v1beta2"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
 
-// Hub marks CloudStackClusterList as a conversion hub.
-func (*CloudStackClusterList) Hub() {}
+const DefaultEndpointCredential = "global"
 
-// Hub marks CloudStackMachine as a conversion hub.
-func (*CloudStackMachine) Hub() {}
+//nolint:golint,revive,stylecheck
+func Convert_v1beta1_CloudStackCluster_To_v1beta2_CloudStackCluster(in *CloudStackCluster, out *v1beta2.CloudStackCluster, s conv.Scope) error {
+	out.ObjectMeta = in.ObjectMeta
+	failureDomains, err := getFailureDomains(in)
+	if err != nil {
+		return err
+	}
+	out.Spec = v1beta2.CloudStackClusterSpec{
+		ControlPlaneEndpoint: in.Spec.ControlPlaneEndpoint,
+		FailureDomains:       failureDomains,
+		Account:              in.Spec.Account,
+		Domain:               in.Spec.Domain,
+	}
 
-// Hub marks CloudStackMachineList as a conversion hub.
-func (*CloudStackMachineList) Hub() {}
+	out.Status = v1beta2.CloudStackClusterStatus{
+		FailureDomains: in.Status.FailureDomains,
+		Ready:          in.Status.Ready,
+	}
+	return nil
+}
+
+//nolint:golint,revive,stylecheck
+func Convert_v1beta2_CloudStackCluster_To_v1beta1_CloudStackCluster(in *v1beta2.CloudStackCluster, out *CloudStackCluster, scope conv.Scope) error {
```

Comment:
Is this function naming convention generally used for conversion webhook?
If not, I'd like to keep the naming style same as others and remove the lint exceptions.

Reply:
yes, this is the standard naming pattern. 

See samples in vsphere: 
https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/main/apis/v1alpha4/vspherecluster_conversion.go